mesonbuild/utils/universal.py

```diff
@@ -1361,34 +1365,33 @@ def do_conf_file(src: str, dst: str, confdata: 'ConfigurationData',
 
 '''
 
-def dump_conf_header(ofilename: str, cdata: 'ConfigurationData', output_format: T.Literal['c', 'nasm']) -> None:
-    if output_format == 'c':
-        prelude = CONF_C_PRELUDE
-        prefix = '#'
-    else:
-        prelude = CONF_NASM_PRELUDE
-        prefix = '%'
+def _dump_c_header(ofile: TextIOWrapper, prelude: str, prefix: str, desc_prefix: str, desc_sep: str, desc_suffix: str, cdata: ConfigurationData) -> None:
```

Comment:
I think this would be a lot simpler if you just handled the output_format in dump_c_header instead of calculating it in dump_conf_header just to pass each bit of data as parameters to another function.